substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1217,6 +1217,41 @@ private int completeImageBuild() {
         }
 
         imageName = getHostedOptionFinalArgumentValue(imageBuilderArgs, oHName);
+        Path imageNamePath;
```

Comment:
Wouldn't it be better to implement theses adjustments in the builder instead of the driver?

Reply:
1. Disallow `-H:Name` to specify absolute path. Relative path is OK (e.g. `foo/bar/myimg`)
2. Perform `mkdir -p` for the `-H:Path` directory
3. Use pathToImage = `-H:Path` + `-H:Name` but if `dirname(pathToImage)` does not exist issue error msg